pkg/handlers/v1/attribute.go

```diff
@@ -0,0 +1,39 @@
+package v1
+
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/assetattributor"
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/domain"
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/logs"
+)
+
+// AttributeHandler handles the Attribution endpoint for nexpose-asset-attributor
+type AttributeHandler struct {
+	AssetAttributor domain.AssetAttributor
+	LogFn           domain.LogFn
+	StatFn          domain.StatFn
+}
+
+// Handle processes the incoming domain.NexposeAssetVulnerabilities instance, queries available asset inventory systems via its
+// AssetAttributor, and returns a domain.AttributedAssetVulnerabilities instance, annotated with the business context of the asset
+// at Nexpose scan time.
+func (h *AttributeHandler) Handle(ctx context.Context, assetVulns domain.NexposeAssetVulnerabilities) (domain.NexposeAttributedAssetVulnerabilities, error) {
+	logger := h.LogFn(ctx)
+
+	attributedAssetVulns, err := h.AssetAttributor.Attribute(ctx, assetVulns)
+	if err != nil {
+		switch err.(type) {
```

Comment:
It doesn't seem like error types result in specific behavior. If your goal of having typed errors is to provide context, I'd recommend looking at https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully.

Otherwise, having typed errors unnecessarily adds to the contract for the function's public interface.

Reply:
I am now using the error types to present the correct status code to callers.